app/queries/wallet_transactions_query.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class WalletTransactionsQuery < BaseQuery
+  def call(wallet_id:, page:, limit:, filters: {})
+    wallet_transactions = base_scope(wallet_id:)
+
+    if valid_transaction_type?(filters[:transaction_type])
+      wallet_transactions = wallet_transactions.where(transaction_type: filters[:transaction_type])
+    end
+    wallet_transactions = wallet_transactions.where(id: filters[:ids]) if filters[:ids].present?
+    wallet_transactions = wallet_transactions.where(status: filters[:status]) if valid_status?(filters[:status])
+    wallet_transactions = wallet_transactions.order(created_at: :desc).page(page).per(limit)
+
+    result.wallet_transactions = wallet_transactions
+    result
+  end
+
+  private
+
+  def base_scope(wallet_id:)
+    Wallet.find(wallet_id).wallet_transactions
```

Comment:
I think it should also be scoped at the organization level as we don't want to expose all wallets :)

Also, maybe the query service should return a `NotFoundFailure` if the wallet is not found.

Reply:
Sure, will fix the first comment; thanks for catching.

Regarding second one, class that is calling query service already rescue `ActiveRecord::RecordNotFound` and return `not_found_error(resource: 'wallet')`

Reply:
The second point is more a suggestion :)

My point is that it could be a way to unify the behavior between resolvers and controllers, as we are doing with services for example.

We could use existing `render_error_response(result)` in controller and `result_error(result)` in resolver, rather than handling a rails error. It would make it easier do handle other types of errors in the future.

Reply:
Refactored :)

Reply:
Great! :rocket: